src/plan_agg_bookend.c

```diff
@@ -484,9 +486,9 @@ find_first_last_aggs_walker(Node *node, List **context)
 			get_opfamily_member(sort_tce->btree_opf, sort_oid, sort_oid, func_strategy->strategy);
 		if (aggsortop == InvalidOid)
 			elog(ERROR,
-				 "Can't resolve sort operator oid for function oid: %d and type: %d",
-				 aggref->aggfnoid,
-				 sort_oid);
+				 "Can't resolve sort operator oid for function \"%s\" and type \"%s\"",
+				 format_procedure(aggref->aggfnoid),
```

Comment:
The result is supposed to be a `palloc`ed string. Should we maybe explicitly deallocate this or is this a non issue with the current memory context?

Reply:
It is a non-issue. All those allocations are on error pathes meaning the memory context will be reset anyway.

Reply:
I guess in many cases, errors return numerical OIDs because the reason for the error was that some of those OIDs didn't correspond to actual types or objects. So, I am wondering if it is a good idea in general to turn lots of error calls into `format_...` instead of printing numerical values. I think in case of `format_procedure` it is handled by that function, but is this always the case?

This is the reason why cache lookup failures print something like:

```
cache lookup failed for attribute %d of relation %u
```
since the whole reason for the cache lookup failure likely was that the relation or attribute didn't exist.

Reply:
actually most of these errors are around datatypes not supported with our functions not about general bogus type ids. e.g. time_bucket with decimal or with domains/custom types